packages/lesswrong/server/callbacks/votingCallbacks.ts

```diff
@@ -130,3 +138,121 @@ postPublishedCallback.add(async (publishedPost: DbPost) => {
   
   await batchUpdateScore({collection: Posts});
 });
+
+// When a vote is cast, if its new karma is above review_market_threshold, create a Manifold
+// on it making top 50 in the review, and create a comment linking to the market.
+
+const manifoldAPIKeySetting = new DatabaseServerSetting<string | null>('manifold.reviewBotKey', null)
+const manifoldAPIKey = manifoldAPIKeySetting.get()
+
+const reviewUserBotSetting = new DatabaseServerSetting<string | null>('reviewBotId', null)
+const reviewUserBot = reviewUserBotSetting.get()
+
+async function addTagToPost(postId: string, tagSlug: string, botUser: DbUser, context: ResolverContext) {
+  const tag = await Tags.findOne({slug: tagSlug})
+  if (!tag) throw new Error(`Tag with slug "${tagSlug}" not found`)
+  await addOrUpvoteTag({tagId: tag._id, postId: postId, currentUser: botUser, context});
+}
+
+// AFAIU the flow, this has a race condition. If a post is voted on twice in quick succession, it will create two markets.
```

Comment:
I think the race condition could be more easily avoided if instead of putting `manifoldReviewMarketId` on the post, you put `postId` on the market (with a unique index constraining it).  Not clear that it's worth refactoring now, but in general parent-child relationships should have the child referencing the parent rather than vice-versa.  (It might not be a parent-child relationship if you strongly anticipate more uses for the new market cache collection than for just review posts, but :shrug:)

Reply:
There is no durable table of markets IIRC

Reply:
Though could refactor to have one